invoice/src/main/java/org/killbill/billing/invoice/InvoiceDispatcher.java

```diff
@@ -836,8 +836,15 @@ private void logInvoiceWithItems(final ImmutableAccountData account, final Invoi
             tmp.append(String.format("Adjusting existing invoiceId='%s', numberOfItems='%d', accountId='%s', targetDate='%s':\n",
                                      adjustedInvoices, invoice.getNumberOfItems(), account.getId(), targetDate));
         }
+        int n = 0;
         for (final InvoiceItem item : invoice.getInvoiceItems()) {
+            if (n > 50) {
```

Comment:
Might be worth extracting as `private final const` and intuitively I would say a default less than 50. Perhaps 20?